src/utils/canvas/generateElement.ts

```diff
@@ -50,11 +50,13 @@ const generateRectElement = (
 ): ZagyCanvasRectElement => {
     const {x,y,endX,endY} =  getCorrectPos(startPos,endPos);
     const opts = normalizeRectOptions(options);
+    const width = Math.max(Math.abs( endPos[0] - startPos[0]),1);
```

Comment:
one problem with this.  variables endX and endY aren't changed to accomodate this change which can result in bugs in places like selection.

Reply:
also this still doesn't solve the problem, and causes another problem... rectangles now can't be drawn upwards.